src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobExecutionWatchdogCommand.php

```diff
@@ -0,0 +1,164 @@
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\PhpExecutableFinder;
+use Symfony\Component\Process\Process;
+
+/**
+ * @author    JM Leroux <jean-marie.leroux@akeneo.com>
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+final class JobExecutionWatchdogCommand extends Command
+{
+    protected static $defaultName = 'akeneo:batch:watchdog';
+    protected static $defaultDescription = '[Internal] Launched by the job queue consumer';
+
+    /** Interval in seconds before updating health check if job is still running. */
+    public const HEALTH_CHECK_INTERVAL = 5;
+
+    /** Interval in microseconds before checking if the process is still running. */
+    private const RUNNING_PROCESS_CHECK_INTERVAL = 200000;
+
+    public function __construct(
+        private JobExecutionManager $executionManager,
+        private LoggerInterface $logger,
+        private string $projectDir
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setHidden(true)
+            ->addArgument('job_execution_id', InputArgument::OPTIONAL, 'Job execution ID')
+            ->addOption(
+                'username',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Username to launch the job instance with'
+            )
+            ->addOption(
+                'email',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'The email to notify at the end of the job execution'
+            )
+            ->addOption(
+                'no-log',
+                null,
+                InputOption::VALUE_NONE,
+                "Don't display logs"
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $pathFinder = new PhpExecutableFinder();
+        $console = sprintf('%s/bin/console', $this->projectDir);
+        $startTime = time();
+        $jobExecutionId = (int)$input->getArgument('job_execution_id');
+        try {
+            $processArguments = $this->buildBatchCommand(
+                $console,
+                $pathFinder->find(),
+                $jobExecutionId,
+                $input->getOptions()
+            );
+            $process = new Process($processArguments);
+            $process->setTimeout(null);
+
+            $this->logger->notice('Launching job execution "{job_execution_id}".', [
+                'job_execution_id' => $jobExecutionId,
+            ]);
+            $this->logger->debug(sprintf('Command line: "%s"', $process->getCommandLine()));
+
+            $this->executeProcess($process, $jobExecutionId);
+        } catch (\Throwable $t) {
+            $this->logger->error(sprintf('An error occurred: %s', $t->getMessage()));
+            $this->logger->error($t->getTraceAsString());
```

Comment:
```suggestion
            $this->logger->error(sprintf('An error occurred: %s', $t->getMessage()), ['exception' => $t]);
```
(let the logger/handler decide if the exception trace should be logged)